llvm/lib/Analysis/VectorUtils.cpp

```diff
@@ -413,6 +413,36 @@ bool llvm::getShuffleDemandedElts(int SrcWidth, ArrayRef<int> Mask,
   return true;
 }
 
+bool llvm::isMaskedSlidePair(ArrayRef<int> Mask, int NumElts,
```

Comment:
Why did this need to move up to VectorUtils? Couldn't it stay in the RISCV backend?

Reply:
I didn't see an obvious spot to put it.  If you have a suggestion, happy to move it back.